src/sentry/utils/outcomes.py

```diff
@@ -32,11 +32,11 @@ def is_billing(self) -> bool:
         return self in (Outcome.ACCEPTED, Outcome.RATE_LIMITED)
 
 
-outcomes_publisher = None
-billing_publisher = None
+T = TypeVar("T")
 
 
-def track_outcome(
+def track_outcome_custom_publish(
+    publish: Callable[[str, str, str], T],
```

Comment:
Isn't the 3rd arg the json payload?

Reply:
It's the JSON payload serialized as string.

Reply:
Oops, thanks

Reply:
I think this could be `publish: Callable[[str], T]` where the arg is the json payload and the topic and cluster name is already resolved by the publish function. It would help simplify this function.